pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,462 @@
+	CheckTestCase(ctx context.Context, jobRuns []jobrunaggregatorapi.JobRunInfo) *junit.TestSuite
+}
+
+type minimumRequiredPassesTestCaseChecker struct {
+	id                     testIdentifier
+	testNameSuffix         string
+	requiredNumberOfPasses int
+}
+
+type testStatus int
+
+const (
+	testSkipped testStatus = iota
+	testPassed
+	testFailed
+)
+
+func getTestStatus(id testIdentifier, testSuite *junit.TestSuite) testStatus {
+	if len(id.testSuites) == 0 || id.testSuites[0] != testSuite.Name {
+		return testSkipped
+	}
+	// We have a top level suite match, search for test case
+	if len(id.testSuites) == 1 {
+		for _, testCase := range testSuite.TestCases {
+			// Exact match will result in either pass or fail
+			if testCase.Name == id.testName {
+				if testCase.FailureOutput == nil {
+					return testPassed
+				} else {
+					return testFailed
+				}
+			}
+		}
+		return testSkipped
+	}
+	// Search next level
+	next := id
+	next.testSuites = id.testSuites[1:]
+	for _, childSuite := range testSuite.Children {
+		if len(next.testSuites) > 0 && next.testSuites[0] == childSuite.Name {
+			status := getTestStatus(next, childSuite)
+			if status == testPassed || status == testFailed {
+				return status
+			}
+		}
+	}
+	return testSkipped
+}
+
+func (r minimumRequiredPassesTestCaseChecker) addTestResultToDetails(currDetails *jobrunaggregatorlib.TestCaseDetails,
+	jobRun jobrunaggregatorapi.JobRunInfo, status testStatus) {
+	switch status {
+	case testPassed:
+		currDetails.Passes = append(
+			currDetails.Passes,
+			jobrunaggregatorlib.TestCasePass{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	case testFailed:
+		currDetails.Failures = append(
+			currDetails.Failures,
+			jobrunaggregatorlib.TestCaseFailure{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	default:
+		currDetails.Skips = append(
+			currDetails.Skips,
+			jobrunaggregatorlib.TestCaseSkip{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	}
+}
+
+// addToTestSuiteFromSuiteNames adds embedded TestSuite from array of suite names to top suite.
+// It returns the bottom level suite for easy access for callers.
+func addToTestSuiteFromSuiteNames(topSuite *junit.TestSuite, suiteNames []string) *junit.TestSuite {
+	previousSuite := topSuite
+	for _, suiteName := range suiteNames {
+		currentSuite := &junit.TestSuite{
+			Name:      suiteName,
+			TestCases: []*junit.TestCase{},
+		}
+		previousSuite.Children = append(previousSuite.Children, currentSuite)
+		previousSuite = currentSuite
+	}
+	return previousSuite
+}
+
+func updateTestCountsInSuite(suite *junit.TestSuite) {
+	for _, test := range suite.TestCases {
+		suite.NumTests++
```

Comment:
please calculate the numbers in a method scoped variable and then do a direct assign.  Done like this, if the method is called twice, the counts will be wrong.